File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
import java.util.Arrays;

public class Solution {
    public int longestConsecutive(int[] nums) {
        Arrays.sort(nums);
        int maxCount = 0;
        int currCount = 1;
        
        if (nums.length == 0) {
            return 0;
        }
        if (nums.length == 1) {
            return 1;
        }
        
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] + 1 == nums[i + 1]) {
                currCount++;
                maxCount = Math.max(maxCount, currCount);
            } else if (nums[i] == nums[i + 1]) {
                maxCount = Math.max(maxCount, currCount);
            } else {
                currCount = 1;
                maxCount = Math.max(maxCount, currCount);
            }
        }
        
        return maxCount;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        if len(nums) == 1:
            return 1
        
        nums.sort()
        maxCount = 1  # Initialize to 1 instead of 0
        currCount = 1
        
        for i in range(len(nums) - 1):
            if nums[i] + 1 == nums[i + 1]:
                currCount += 1
                maxCount = max(maxCount, currCount)
            elif nums[i] == nums[i + 1]:
                continue  # Skip duplicates
            else:
                currCount = 1
                # No need to update maxCount here
        
        return maxCount
        